petastorm/py_dict_reader_worker.py

```diff
@@ -34,6 +34,24 @@ def _merge_two_dicts(a, b):
     return result
 
 
+def _apply_transform_spec(all_rows, transform_spec):
+    """Applies transform_spec.func to all rows in all_rows (if func is specified). Removes all fields as specified by
+    transform_spec.removed_fields, unless the fields were already deleted by transform_spec.func.
+
+    All modifications are performed in-place.
+    """
+    if transform_spec.func:
+        all_rows = [transform_spec.func(row) for row in all_rows]
+
+    if transform_spec.removed_fields:
+        for row in all_rows:
```

Comment:
I was actually wondering if `all_rows` have a method is a dict or have a method that can fetch all their columns? For eg if it has `all_rows.columns` or `all_rows.keys()`. If it does then we could just do something like

```python
for field_to_remove  in (all_rows.keys() & transform_spec.removed_fields):
    del row[field_to_remove]
```